### DISCUSSION TITLE
Building a open source changelog index

### DISCUSSION THREAD

**Discussion by jayfk** [olivierlacan/keep-a-changelog#484] [Tools]
I recently open sourced the changelog finder/parser that's used on [pyup.io](https://pyup.io) here: [pyupio/changelogs](https://github.com/pyupio/changelogs)

It's basically a library with a CLI that tries to find changelogs for packages on PyPi, npm and rubygems. It works relatively well, has a huge testing library for all kinds of different changelogs and is designed in a way that makes it easy to add new package indexes.

I want to build a fully open service on top of it that finds and parses changelogs for any given library, provides an API and maybe a badge service.

Is anyone interested on working on this with me? I'd probably sponsor infrastructure from pyup.io and the changelogs.net domain to get started.

I have a couple of ideas on how to build the service (because I'll probably write most of it :D), but I'm open to any suggestions.

**Comment by kavdev** [comment]
@jayfk: If it's written in Django, hosted on GitHub, and you put those ideas in to issues, I'd be happy to contribute when I find some free time.

**Comment by svetlyak40wt** [comment]
Interesting to see yet another project about changelogs gathering. I've build one too (https://allmychanges.com) and it has very advanced parser which is able to work with plain-text, markdown, restructured and almost any html, but sometimes requires a manual tuning. However, this parser is not extracted as a library and I don't see why somebody need it as a library or a command line utility.

Why do you think it will be useful, and what are the major use cases?

**Comment by kavdev** [comment]
@svetlyak40wt: I personally think we need a pep for changelog format.

Until that happens, we might as well have everyone that's working on changelog parsing put their work into a single, open source codebase. That way different services can use it to enhance their products (eg [Warehouse](https://pypi.org/) and [pyup.io](https://pyup.io)) and we have one (sort-of-standard) body of work that everyone can improve upon.

**Comment by jayfk** [comment]
> I personally think we need a pep for changelog format.

At this point we probably need a full blown RFC for the whole mess that are called changelogs :)



> However, this parser is not extracted as a library and I don't see why somebody need it as a library or a command line utility.

This can be useful if you are doing packaging for a distro, doing project reviews, updates or security research to name a few.

But I really don't want to spend too much time discussing how something like this can be useful. I'd much rather use the time to build it.

@svetlyak40wt: Is the code on https://github.com/AllMyChanges/allmychanges.com still the latest?

**Comment by svetlyak40wt** [comment]
@jayfk what is the difference between getting information from central place like pyup.io or allmychanges.com and parsing it with a command line utility?
